sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryContentClient.java

```diff
@@ -536,10 +535,9 @@ private Context downloadBlobInternal(String digest, WritableByteChannel channel,
             // TODO (limolkova) https://github.com/Azure/azure-sdk-for-java/issues/34400
             context = context.addData("azure-eagerly-read-response", true);
             Response<BinaryData> lastChunk = blobsImpl.getChunkWithResponse(repositoryName, digest, range.toString(), context);
-            validateResponseHeaderDigest(digest, lastChunk.getHeaders());
+            long blobSize = getBlobSize(lastChunk.getHeaders().get(HttpHeaderName.CONTENT_RANGE));
```

Comment:
There is a utility method `CoreUtils.extractSizeFromContentRange` that this can be replaced with.